pkg/thing/delivery/http/thing_proxy.go

```diff
@@ -267,7 +267,7 @@ func (p proxy) GetThing(authorization, ID string) (*entities.Thing, error) {
 	for i := range things {
 		t := things[i]
 		if t.Metadata.Knot.ID == ID {
-			nt := &entities.Thing{ID: t.ID, Name: t.Name, Schema: t.Metadata.Knot.Schema}
+			nt := &entities.Thing{ID: ID, Token: t.ID, Name: t.Name, Schema: t.Metadata.Knot.Schema}
```

Comment:
Would **line 210** also need this modification? :thinking: 
> **210** `things = append(things, &entities.Thing{ID: t.Metadata.Knot.ID, Name: t.Name, Schema: t.Metadata.Knot.Schema})`

Reply:
Not in this case @lcbm. Note that the `things` variable will be returned to the user as response of a `device.cmd.list` request, and he doesn't need to know the token (mainflux user's service ID) of every device. Also, the KNoT ID value is right (accessed via Metadata).

Reply:
Gotcha! Thanks for the thorough explanation :smile: